lib/auth/auth.go

```diff
@@ -170,19 +168,19 @@ func (a *AuthServer) GetLocalDomain() (string, error) {
 
 // GenerateHostCert generates host certificate, it takes pkey as a signing
 // private key (host certificate authority)
-func (s *AuthServer) GenerateHostCert(key []byte, hostID, authDomain string, role teleport.Role, ttl time.Duration) ([]byte, error) {
+func (s *AuthServer) GenerateHostCert(key []byte, hostID, authDomain string, roles teleport.Roles, ttl time.Duration) ([]byte, error) {
 	ca, err := s.Trust.GetCertAuthority(services.CertAuthID{
 		Type:       services.HostCA,
 		DomainName: s.DomainName,
 	}, true)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, trace.Errorf("failed to load host CA for '%s': %v", s.DomainName, err)
```

Comment:
seems like a downgrade since we are loosing error type here. Please return appropriate error here